src/edu/ucsd/library/xdre/tab/ExcelSource.java

```diff
@@ -248,23 +254,49 @@ else if ( cache != null && cache.size() > 0)
 	                    List<String> validValue = CONTROL_VALUES.get(header);
 	                    if (validValue != null && validValue.size() > 0) 
```

Comment:
Possible to just return here if `validValue` is `null`? Might let you collapse one conditional path at least.